src/lib/mongoBackend/mongoRegistrationGet.cpp

```diff
@@ -0,0 +1,325 @@
+/*
+*
+* Copyright 2017 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <string>
+#include <vector>
+
+#include "mongo/client/dbclient.h"
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+#include "common/sem.h"
+#include "common/statistics.h"
+#include "common/errorMessages.h"
+#include "rest/OrionError.h"
+#include "rest/HttpStatusCode.h"
+#include "apiTypesV2/Registration.h"
+#include "mongoBackend/dbConstants.h"
+#include "mongoBackend/safeMongo.h"
+#include "mongoBackend/MongoGlobal.h"
+#include "mongoBackend/connectionOperations.h"
+#include "mongoBackend/mongoRegistrationGet.h"
+
+
+
+/* ****************************************************************************
+*
+* setRegistrationId -
+*/
+static void setRegistrationId(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->id = getFieldF(r, "_id").OID().toString();
+}
+
+
+
+/* ****************************************************************************
+*
+* setDescription -
+*/
+static void setDescription(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  if (r.hasField(REG_DESCRIPTION))
+  {
+    regP->description         = getStringFieldF(r, REG_DESCRIPTION);
+    regP->descriptionProvided = true;
+  }
+  else
+  {
+    regP->description         = "";
+    regP->descriptionProvided = false;
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setProvider -
+*/
+static void setProvider(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->provider.url = (r.hasField(REG_PROVIDING_APPLICATION))? getStringFieldF(r, REG_PROVIDING_APPLICATION): "";
+}
+
+
+
+/* ****************************************************************************
+*
+* setDataProvided -
+*
+* Make sure there is only ONE "contextRegistration" in the vector
+* If we have more than one, then the Registration is made in API V1 as this is not
+* possible in V2 and we cannot respond to the request using the current implementation of V2
```

Comment:
I'd suggest to add a reference to issue https://github.com/telefonicaid/fiware-orion/issues/3044, i.e.

> This function will be change to work in different way upon completion of issue #3044

or something similar

Reply:
Fixed in f114c3e